app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java

```diff
@@ -22,33 +24,21 @@
     private String firstOperand = EMPTY_STRING;
     private String secondOperand = EMPTY_STRING;
     private int switchOp = FIRST_OPERAND;
-
-    private void switchOperand() {
-        if (switchOp == FIRST_OPERAND) {
-            switchOp = SECOND_OPERAND;
-        } else {
-            switchOp = FIRST_OPERAND;
-        }
-
-    }
+    private Error error = Error.NONE;
 
     private void addOperand(String value) {
-        switch (switchOp) {
-            case FIRST_OPERAND: {
-                if (!firstOperand.equals(ZERO_RESULT_STRING)) {
-                    firstOperand += value;
-                } else {
-                    firstOperand = value;
-                }
-                break;
+        if (switchOp == FIRST_OPERAND) {
+            if (firstOperand.isEmpty()) {
```

Comment:
can you ommit this if lofig and just add firstOperand += value?

Reply:
Solved!